### DISCUSSION TITLE
How to do this?

### DISCUSSION THREAD

**Discussion by pushuo** [fabricjs/fabric.js#7001] [Q&A]
## Version
4.3.1

## Information about environment
browser

#### Can you give me a hint
![68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f524d34624e4f573635325566593037506a5a2f67697068792e676966](https://user-images.githubusercontent.com/21018378/113966244-4f9ef800-9861-11eb-82cc-8d6ae4911b5b.gif)

**Comment by melchiar** [comment]
The example you've shown would likely require masking to show the outside image at a lower opacity. Although masking is not officially supported, there is an experimental masking implementation here you could try https://github.com/fabricjs/fabric.js/pull/6310

If you can get the masking working its just a clever combination of events to constrain the image's movements to the cropping rectangle.

As a separate note I'd recommend choosing a more relevant name for your issue so that it can be helpful to others searching for the same thing (eg. "Image cropping within a masked container")

**Reply by pushuo** [reply]
Yes, I actually want to restrict the movement of an object
 I want to ask this question: https://github.com/fabricjs/fabric.js/issues/6585。 

What to do? thank.

**Reply by pushuo** [reply]
Or should I reopen a discussion?

**Reply by melchiar** [reply]
@pushuo you can change the name of this discussion. You do not need to open another one
